pkg/monitoring/prometheusagent/secret.go

```diff
@@ -1,27 +1,65 @@
 package prometheusagent
 
 import (
+	"context"
 	"fmt"
+	"strings"
 
 	"github.com/pkg/errors"
 	promv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 	"sigs.k8s.io/yaml"
+)
 
-	"github.com/giantswarm/observability-operator/pkg/monitoring"
+const (
+	authSecretName      = "mimir-basic-auth"
+	authSecretNamespace = "mimir"
+	remoteWriteName     = "mimir"
 )
 
+func GetMimirIngressPassword(ctx context.Context) (string, error) {
```

Comment:
Maybe this function should be in the mimir package and this rely on mimir instead or mimir rely on this?

Reply:
That's what I did first but this came with import issues on the prometheusagent package :/